src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/endtoendtests/PaymentsTest.java

```diff
@@ -0,0 +1,148 @@
+            .formParam("client_id", "bsp")
+            .formParam("client_secret", idamClientSecret)
+            .formParam("scope", "openid profile roles")
+            .formParam("username", username)
+            .formParam("password", password)
+            .post("/o/token");
+
+        assertThat(idamResponse.getStatusCode()).isEqualTo(SC_OK);
+
+        Map<?, ?> r = objectMapper.readValue(idamResponse.getBody().print(), Map.class);
+        return (String)r.get("access_token");
+    }
+
+    private String getS2SToken() throws IOException {
+        String s2sUrl = conf.getString("s2s-url");
+        String s2sSecret = conf.getString("s2s-secret");
+        final String oneTimePassword = format("%06d", new GoogleAuthenticator().getTotpPassword(s2sSecret));
+        Map<String, String> signInDetails = new HashMap<>();
+        signInDetails.put("microservice", "bulk_scan_orchestrator");
+        signInDetails.put("oneTimePassword", oneTimePassword);
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        objectMapper.writeValue(baos, signInDetails);
+        String signInDetailsStr = baos.toString();
+
+        Response s2sResponse = RestAssured
+            .given()
+            .relaxedHTTPSValidation()
+            .baseUri(s2sUrl)
+            .header(CONTENT_TYPE, APPLICATION_JSON.getMimeType())
+            .body(signInDetailsStr)
+            .post("/lease");
+
+        assertThat(s2sResponse.getStatusCode()).isEqualTo(SC_OK);
+
+        return s2sResponse.getBody().print();
+    }
+
+    private Map<?, ?> getCaseData(
+        String accessToken,
+        String s2sToken,
+        String ccdId
+    ) throws JsonProcessingException {
+        String coreCaseDataApiUrl = conf.getString("core-case-data-api-url");
+        Response caseResponse = RestAssured
+            .given()
+            .relaxedHTTPSValidation()
+            .baseUri(coreCaseDataApiUrl)
+            .header("experimental", true)
+            .header("Authorization", BEARER + accessToken)
+            .header("ServiceAuthorization", BEARER + s2sToken)
+            .get("/cases/" + ccdId);
+
+        assertThat(caseResponse.getStatusCode()).isEqualTo(SC_OK);
+
+        Map<?, ?> c = objectMapper.readValue(caseResponse.getBody().print(), Map.class);
+        return (Map<?, ?>) c.get("data");
+    }
+
+    private String assertCompletedProcessorResultAndRetrieveCcdId(String zipFileName) {
+        assertThat(getZipFileStatus(zipFileName)).hasValueSatisfying(env -> {
+            assertThat(env.ccdId).isNotBlank();
+            assertThat(env.container).isEqualTo(Container.BULKSCAN.name);
+            assertThat(env.envelopeCcdAction).isEqualTo("EXCEPTION_RECORD");
+            assertThat(env.id).isNotBlank();
+            assertThat(env.status).isEqualTo("COMPLETED");
+        });
+
+        ProcessorEnvelopeResult processorEnvelopeResult = getZipFileStatus(zipFileName).get();
+        return processorEnvelopeResult.ccdId;
```

Comment:
this can be done outside this method not sure why this and above assertions are combined

Reply:
Done